src/sentry/workflow_engine/types.py

```diff
@@ -32,17 +33,19 @@ class DetectorPriorityLevel(IntEnum):
 ProcessedDataConditionResult = tuple[bool, list[DataConditionResult]]
 
 
-class EventJob(TypedDict):
+@dataclass(frozen=True)
+class EventJob:
     event: GroupEvent
 
 
-class WorkflowJob(EventJob, total=False):
-    group_state: GroupState
-    is_reprocessed: bool
-    has_reappeared: bool
-    has_alert: bool
-    has_escalated: bool
-    workflow: Workflow
+@dataclass(frozen=True)
+class WorkflowJob(EventJob):
+    group_state: GroupState | None = None
+    is_reprocessed: bool | None = None
+    has_reappeared: bool | None = None
```

Comment:
is there a difference between `has_reappeared` and `group_state["is_regression"]`?

Reply:
`has_reappeared` is related to unresolved --> resolved
https://github.com/getsentry/sentry/blob/6cea12f09116576aafb4e003226b94c8b9c1c0b0/src/sentry/tasks/post_process.py#L787-L788

`is_regression` seems to be calculated in a different way
https://github.com/getsentry/sentry/blob/6cea12f09116576aafb4e003226b94c8b9c1c0b0/src/sentry/event_manager.py#L1339-L1344